--- 389-ds-base.spec.orig	2010-10-27 02:46:18.000000000 -0400
+++ 389-ds-base.spec	2010-10-27 02:48:04.000000000 -0400
@@ -12,7 +12,7 @@
 Summary:          389 Directory Server (base)
 Name:             389-ds-base
 Version:          1.2.6.1
-Release:          %{?relprefix}2%{?prerel}%{?dist}
+Release:          %{?relprefix}2%{?prerel}%{?dist}.scripts.%{scriptsversion}
 License:          GPLv2 with exceptions
 URL:              http://port389.org/
 Group:            System Environment/Daemons
@@ -87,6 +87,9 @@
 Source1:          %{name}-git.sh
 Source2:          %{name}-devel.README
 
+Provides:         scripts-389-ds-base
+Patch1000:        389-sasl-io-4-bytes.patch
+
 %description
 389 Directory Server is an LDAPv3 compliant server.  The base package includes
 the LDAP server and command line utilities for server administration.
@@ -113,6 +116,8 @@
 %setup -q -n %{name}-%{version}%{?prerel}
 cp %{SOURCE2} README.devel
 
+%patch1000 -p1 -b .sasl-io
+
 %build
 %configure --enable-autobind --with-selinux
 
@@ -347,6 +352,9 @@
 %{_datadir}/%{pkgname}-selinux
 
 %changelog
+* Wed Oct 27 2010 Geoffrey Thomas <geofft@mit.edu> - 1.2.6.1-2.scripts
+- Cherry-pick patch 310c056 from git to address RHBZ#637852
+
 * Wed Sep 29 2010 Rich Megginson <rmeggins@redhat.com> - 1.2.6.1-2
 - bump rev to rebuild
 
