--- 389-ds-base.spec.orig	2010-12-18 01:18:33.000000000 -0500
+++ 389-ds-base.spec	2010-12-18 01:37:42.000000000 -0500
@@ -13,7 +13,7 @@
 Summary:          389 Directory Server (base)
 Name:             389-ds-base
 Version:          1.2.7.4
-Release:          %{?relprefix}1%{?prerel}%{?dist}
+Release:          %{?relprefix}1%{?prerel}.scripts.%{scriptsversion}%{?dist}
 License:          GPLv2 with exceptions
 URL:              http://port389.org/
 Group:            System Environment/Daemons
@@ -88,6 +88,8 @@
 Source1:          %{name}-git.sh
 Source2:          %{name}-devel.README
 
+Patch0:           389-ds-base-fix-changelog-paths.patch
+
 %description
 389 Directory Server is an LDAPv3 compliant server.  The base package includes
 the LDAP server and command line utilities for server administration.
@@ -102,6 +104,7 @@
 Requires:         mozldap-devel
 %endif
 Provides:         fedora-ds-base-devel = %{version}-%{release}
+Provides:         scripts-389-ds
 Obsoletes:        fedora-ds-base-devel < 1.2.1-1
 
 %description      devel
@@ -109,6 +112,7 @@
 
 %prep
 %setup -q -n %{name}-%{version}%{?prerel}
+%patch0 -p1 -b .fix-changelog-paths
 cp %{SOURCE2} README.devel
 
 %build
@@ -304,6 +308,9 @@
 - 1.2.7.4 release - git tag 389-ds-base-1.2.7.4
 - Bug 661792 - Valid managed entry config rejected
 
+* Thu Dec  9 2010 Mitchell E Berger <mitchb@mit.edu> - 1.2.7.1-1.scripts
+- Correct changelog path logic in MMR plugin
+
 * Wed Dec  8 2010 Rich Megginson <rmeggins@redhat.com> - 1.2.7.3-1
 - 1.2.7.3 release - git tag 389-ds-base-1.2.7.3
 - Bug 658312 - Invalid free in Managed Entry plug-in
