--- 389-ds-base.spec.orig	2010-12-16 12:22:05.000000000 -0500
+++ 389-ds-base.spec	2010-12-23 21:19:49.000000000 -0500
@@ -13,7 +13,7 @@
 Summary:          389 Directory Server (base)
 Name:             389-ds-base
 Version:          1.2.7.5
-Release:          %{?relprefix}1%{?prerel}%{?dist}
+Release:          %{?relprefix}1%{?prerel}.scripts.%{scriptsversion}%{?dist}
 License:          GPLv2 with exceptions
 URL:              http://port389.org/
 Group:            System Environment/Daemons
@@ -83,11 +83,15 @@
 Requires(preun): /sbin/chkconfig
 Requires(preun): /sbin/service
 
+Provides:         scripts-389-ds
+
 Source0:          http://port389.org/sources/%{name}-%{version}%{?prerel}.tar.bz2
 # 389-ds-git.sh should be used to generate the source tarball from git
 Source1:          %{name}-git.sh
 Source2:          %{name}-devel.README
 
+Patch0:           389-ds-base-fix-changelog-paths.patch
+
 %description
 389 Directory Server is an LDAPv3 compliant server.  The base package includes
 the LDAP server and command line utilities for server administration.
@@ -109,6 +113,7 @@
 
 %prep
 %setup -q -n %{name}-%{version}%{?prerel}
+%patch0 -p1 -b .fix-changelog-paths
 cp %{SOURCE2} README.devel
 
 %build
@@ -300,6 +305,9 @@
 %{_libdir}/%{pkgname}/*.so
 
 %changelog
+* Thu Dec 23 2010 Mitchell E Berger <mitchb@mit.edu> - 1.2.7.5-1.scripts
+- Correct changelog path logic in MMR plugin
+
 * Thu Dec 16 2010 Rich Megginson <rmeggins@redhat.com> - 1.2.7.5-1
 - 1.2.7.5 release - git tag 389-ds-base-1.2.7.5
 - Bug 663597 - Memory leaks in normalization code
