--- gnutls.spec.orig	2014-03-08 16:13:24.922925743 -0500
+++ gnutls.spec	2014-03-08 16:14:50.464231133 -0500
@@ -1,7 +1,7 @@
 Summary: A TLS protocol implementation
 Name: gnutls
 Version: 2.12.23
-Release: 2%{?dist}
+Release: 2%{?dist}.scripts.%{scriptsversion}
 # The libgnutls library is LGPLv2+, utilities and remaining libraries are GPLv3+
 License: GPLv3+ and LGPLv2+
 Group: System Environment/Libraries
@@ -26,6 +26,8 @@
 # Use only FIPS approved ciphers in the FIPS mode
 Patch7: gnutls-2.12.21-fips-algorithms.patch
 Patch8: gnutls-2.12.23-cve-2013-2116.patch
+# Patch GNUTLS-SA-2014-2
+Patch9: gnutls-2.12.x-cve-2014-0092.patch
 
 BuildRoot:  %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 Requires: libgcrypt >= 1.2.2
@@ -99,6 +101,7 @@
 %patch6 -p1 -b .cli-debug
 %patch7 -p1 -b .fips
 %patch8 -p1 -b .overread
+%patch9 -p1 -b .cve-2014-0092
 
 for i in auth_srp_rsa.c auth_srp_sb64.c auth_srp_passwd.c auth_srp.c gnutls_srp.c ext_srp.c; do
     touch lib/$i
