--- httpd.spec.~1~	2012-02-13 09:46:36.000000000 -0500
+++ httpd.spec	2012-03-26 01:33:53.247271289 -0400
@@ -8,7 +8,7 @@
 Summary: Apache HTTP Server
 Name: httpd
 Version: 2.2.22
-Release: 1%{?dist}
+Release: 1%{?dist}.scripts.%{scriptsversion}
 URL: http://httpd.apache.org/
 Source0: http://www.apache.org/dist/httpd/httpd-%{version}.tar.bz2
 Source1: index.html
@@ -54,6 +54,15 @@
 Provides: httpd-mmn = %{mmn}, httpd-mmn = %{mmnisa}
 Requires: httpd-tools = %{version}-%{release}, apr-util-ldap, systemd-units
 
+Provides: scripts-httpd = %{version}-%{release}
+Patch1000: httpd-suexec-scripts.patch
+Patch1003: httpd-2.2.x-mod_status-security.patch
+Patch1004: httpd-2.2.x-304.patch
+Patch1005: httpd-2.2.x-mod_ssl-sessioncaching.patch
+Patch1006: httpd-suexec-cloexec.patch
+Patch1007: httpd-fixup-vhost.patch
+Patch1008: httpd-sysv-deps.patch
+
 %description
 The Apache HTTP Server is a powerful, efficient, and extensible
 web server.
@@ -64,6 +76,7 @@
 Obsoletes: secureweb-devel, apache-devel, stronghold-apache-devel
 Requires: apr-devel, apr-util-devel, pkgconfig
 Requires: httpd = %{version}-%{release}
+Provides: scripts-httpd-devel = %{version}-%{release}
 
 %description devel
 The httpd-devel package contains the APXS binary and other files
@@ -102,6 +115,7 @@
 Requires(post): openssl, /bin/cat
 Requires(pre): httpd
 Requires: httpd = 0:%{version}-%{release}, httpd-mmn = %{mmnisa}
+Provides: scripts-mod_ssl
 Obsoletes: stronghold-mod_ssl
 
 %description -n mod_ssl
@@ -110,6 +124,11 @@
 Security (TLS) protocols.
 
 %prep
+
+# Horrible hack to patch the httpd.init file
+cd $RPM_SOURCE_DIR
+%patch1008 -p1 -b .sysv-deps
+
 %setup -q
 %patch1 -p1 -b .apctl
 %patch2 -p1 -b .apxs
@@ -128,6 +147,14 @@
 # Patch in vendor/release string
 sed "s/@RELEASE@/%{vstring}/" < %{PATCH20} | patch -p1
 
+%patch1000 -p1 -b .scripts
+%patch1003 -p1 -b .permitstatus
+%patch1004 -p1 -b .scripts-304
+%patch1005 -p1 -b .ssl-sessioncache
+%patch1006 -p1 -b .cloexec
+%patch1007 -p1 -b .fixup-vhost
+# Note that patch1008 is not here, as it patches the initscript elsewhere in this .spec
+
 # Safety check: prevent build if defined MMN does not equal upstream MMN.
 vmmn=`echo MODULE_MAGIC_NUMBER_MAJOR | cpp -include include/ap_mmn.h | sed -n '/^2/p'`
 if test "x${vmmn}" != "x%{mmn}"; then
@@ -175,10 +205,12 @@
         --with-apr=%{_prefix} --with-apr-util=%{_prefix} \
 	--enable-suexec --with-suexec \
 	--with-suexec-caller=%{suexec_caller} \
-	--with-suexec-docroot=%{contentdir} \
+	--with-suexec-docroot=/ \
+	--with-suexec-userdir=web_scripts \
+	--with-suexec-trusteddir=/usr/libexec/scripts-trusted \
 	--with-suexec-logfile=%{_localstatedir}/log/httpd/suexec.log \
 	--with-suexec-bin=%{_sbindir}/suexec \
-	--with-suexec-uidmin=500 --with-suexec-gidmin=100 \
+	--with-suexec-uidmin=50 --with-suexec-gidmin=50 \
         --enable-pie \
         --with-pcre \
 	$*
