--- httpd.spec.orig	2009-03-28 13:32:47.000000000 -0400
+++ httpd.spec	2009-03-28 14:20:15.000000000 -0400
@@ -6,7 +6,7 @@
 Summary: Apache HTTP Server
 Name: httpd
 Version: 2.2.10
-Release: 1%{?dist}
+Release: 1%{?dist}.scripts.%{scriptsversion}
 URL: http://httpd.apache.org/
 Source0: http://www.apache.org/dist/httpd/httpd-%{version}.tar.gz
 Source1: index.html
@@ -54,6 +54,12 @@
 Conflicts: pcre < 4.0
 Requires: httpd-tools = %{version}-%{release}
 
+Provides: scripts-httpd
+Patch1000: httpd-suexec-scripts.patch
+Patch1001: httpd-2.2.x-sni.patch
+Patch1003: httpd-2.2.x-mod_status-security.patch
+Patch1004: httpd-2.2.x-304.patch
+
 %description
 The Apache HTTP Server is a powerful, efficient, and extensible
 web server.
@@ -100,6 +106,7 @@
 BuildRequires: openssl-devel, distcache-devel
 Requires(post): openssl >= 0.9.7f-4, /bin/cat
 Requires: httpd = 0:%{version}-%{release}, httpd-mmn = %{mmn}
+Provides: scripts-mod_ssl
 Obsoletes: stronghold-mod_ssl
 
 %description -n mod_ssl
@@ -127,6 +134,11 @@
 # Patch in vendor/release string
 sed "s/@RELEASE@/%{vstring}/" < %{PATCH20} | patch -p1
 
+%patch1000 -p1 -b .scripts
+%patch1001 -p1 -b .sni
+%patch1003 -p1 -b .permitstatus
+%patch1004 -p1 -b .scripts-304
+
 # Safety check: prevent build if defined MMN does not equal upstream MMN.
 vmmn=`echo MODULE_MAGIC_NUMBER_MAJOR | cpp -include include/ap_mmn.h | sed -n '/^2/p'`
 if test "x${vmmn}" != "x%{mmn}"; then
@@ -175,10 +187,12 @@
         --with-apr=%{_prefix} --with-apr-util=%{_prefix} \
 	--enable-suexec --with-suexec \
 	--with-suexec-caller=%{suexec_caller} \
-	--with-suexec-docroot=%{contentdir} \
+	--with-suexec-docroot=/ \
+	--with-suexec-userdir=web_scripts \
+	--with-suexec-trusteddir=/usr/libexec/scripts-trusted \
 	--with-suexec-logfile=%{_localstatedir}/log/httpd/suexec.log \
 	--with-suexec-bin=%{_sbindir}/suexec \
-	--with-suexec-uidmin=500 --with-suexec-gidmin=100 \
+	--with-suexec-uidmin=50 --with-suexec-gidmin=50 \
         --enable-pie \
         --with-pcre \
 	$*
