--- krb5.spec.orig	2010-01-12 14:27:12.000000000 -0500
+++ krb5.spec	2010-01-13 04:22:13.000000000 -0500
@@ -16,7 +16,7 @@
 Summary: The Kerberos network authentication system.
 Name: krb5
 Version: 1.6.3
-Release: 23%{?dist}
+Release: 23%{?dist}.scripts.%{scriptsversion}
 # Maybe we should explode from the now-available-to-everybody tarball instead?
 # http://web.mit.edu/kerberos/dist/krb5/1.6/krb5-1.6.2-signed.tar
 Source0: krb5-%{version}.tar.gz
@@ -107,6 +107,8 @@
 Patch84: krb5-CVE-2009-0847.patch
 Patch85: http://web.mit.edu/kerberos/advisories/2009-004-patch_1.6.3.txt
 
+Patch1000: krb5-kuserok-scripts.patch
+
 License: MIT
 URL: http://web.mit.edu/kerberos/www/
 Group: System Environment/Libraries
@@ -148,6 +150,7 @@
 Group: System Environment/Libraries
 Prereq: grep, /sbin/ldconfig, sh-utils
 Obsoletes: krb5-configs
+Provides: scripts-krb5-libs
 
 %description libs
 Kerberos is a network authentication system. The krb5-libs package
@@ -1428,6 +1431,7 @@
 %patch83 -p1 -b .CVE-2009-0846
 %patch84 -p1 -b .CVE-2009-0847
 %patch85 -p0 -b .2009-004
+%patch1000 -p1 -b .kuserok
 cp src/krb524/README README.krb524
 gzip doc/*.ps
 
