--- krb5.spec.orig	2010-04-06 14:47:17.000000000 -0400
+++ krb5.spec	2010-04-11 05:51:33.000000000 -0400
@@ -16,7 +16,7 @@
 Summary: The Kerberos network authentication system.
 Name: krb5
 Version: 1.6.3
-Release: 29%{?dist}
+Release: 29%{?dist}.scripts.%{scriptsversion}
 # Maybe we should explode from the now-available-to-everybody tarball instead?
 # http://web.mit.edu/kerberos/dist/krb5/1.6/krb5-1.6.2-signed.tar
 Source0: krb5-%{version}.tar.gz
@@ -113,6 +113,8 @@
 Patch87: krb5-1.6.3-kpasswd_ipv6.patch
 Patch88: krb5-1.6.1-cs22427.patch
 
+Patch1000: krb5-kuserok-scripts.patch
+
 License: MIT
 URL: http://web.mit.edu/kerberos/www/
 Group: System Environment/Libraries
@@ -154,6 +156,7 @@
 Group: System Environment/Libraries
 Prereq: grep, /sbin/ldconfig, sh-utils
 Obsoletes: krb5-configs
+Provides: scripts-krb5-libs
 
 %description libs
 Kerberos is a network authentication system. The krb5-libs package
@@ -1469,6 +1472,7 @@
 %patch86 -p0 -b .openssl-1.0
 %patch87 -p0 -b .kpasswd_ipv6
 %patch88 -p0 -b .cs22427
+%patch1000 -p1 -b .kuserok
 cp src/krb524/README README.krb524
 gzip doc/*.ps
 
