--- krb5.spec.orig	2012-06-17 19:29:24.140005717 -0400
+++ krb5.spec	2012-06-17 19:30:47.834405053 -0400
@@ -20,7 +20,7 @@
 Summary: The Kerberos network authentication system
 Name: krb5
 Version: 1.10.2
-Release: 2%{?dist}
+Release: 2%{?dist}.scripts.%{scriptsversion}
 # Maybe we should explode from the now-available-to-everybody tarball instead?
 # http://web.mit.edu/kerberos/dist/krb5/1.10/krb5-1.10.2-signed.tar
 Source0: krb5-%{version}.tar.gz
@@ -69,6 +69,8 @@
 Patch105: krb5-kvno-230379.patch
 Patch106: krb5-1.10.2-keytab-etype.patch
 
+Patch1000: krb5-kuserok-scripts.patch
+
 License: MIT
 URL: http://web.mit.edu/kerberos/www/
 Group: System Environment/Libraries
@@ -126,6 +128,7 @@
 %package libs
 Summary: The shared libraries used by Kerberos 5
 Group: System Environment/Libraries
+Provides: scripts-krb5-libs, scripts-krb5-libs%{?_isa}
 
 %description libs
 Kerberos is a network authentication system. The krb5-libs package
@@ -245,6 +248,7 @@
 %patch103 -p0 -b .gcc47
 %patch105 -p1 -b .kvno
 %patch106 -p1 -b .keytab-etype
+%patch1000 -p1 -b .kuserok
 rm src/lib/krb5/krb/deltat.c
 
 gzip doc/*.ps
