--- krb5.spec.~1~	2011-12-06 14:15:43.000000000 -0500
+++ krb5.spec	2012-01-31 17:31:25.770200775 -0500
@@ -6,7 +6,7 @@
 Summary: The Kerberos network authentication system
 Name: krb5
 Version: 1.9.2
-Release: 4%{?dist}
+Release: 4%{?dist}.scripts.%{scriptsversion}
 # Maybe we should explode from the now-available-to-everybody tarball instead?
 # http://web.mit.edu/kerberos/dist/krb5/1.9/krb5-1.9.1-signed.tar
 Source0: krb5-%{version}.tar.gz
@@ -62,6 +62,8 @@
 Patch90: krb5-1.9-aes-hmac.patch
 Patch91: http://web.mit.edu/kerberos/advisories/2011-007-patch.txt
 
+Patch1000: krb5-kuserok-scripts.patch
+
 License: MIT
 URL: http://web.mit.edu/kerberos/www/
 Group: System Environment/Libraries
@@ -111,6 +113,7 @@
 %package libs
 Summary: The shared libraries used by Kerberos 5
 Group: System Environment/Libraries
+Provides: scripts-krb5-libs, scripts-krb5-libs%{?_isa}
 
 %description libs
 Kerberos is a network authentication system. The krb5-libs package
@@ -215,6 +218,7 @@
 %patch89 -p1 -b .sendto_poll3
 %patch90 -p1 -b .aes-hmac
 %patch91 -p1 -b .2011-007
+%patch1000 -p1 -b .kuserok
 gzip doc/*.ps
 
 sed -i -e '1s!\[twoside\]!!;s!%\(\\usepackage{hyperref}\)!\1!' doc/api/library.tex
