--- krb5.spec.~1~	2019-05-26 00:04:40.000000000 -0400
+++ krb5.spec	2019-05-26 00:07:14.000000000 -0400
@@ -18,7 +18,7 @@
 Name: krb5
 Version: 1.17
 # for prerelease, should be e.g., 0.% {prerelease}.1% { ?dist } (without spaces)
-Release: 14%{?dist}
+Release: 14%{?dist}.scripts.%{scriptsversion}
 
 # lookaside-cached sources; two downloads and a build artifact
 Source0: https://web.mit.edu/kerberos/dist/krb5/1.16/krb5-%{version}%{prerelease}.tar.gz
@@ -88,6 +88,8 @@
 Patch116: Clear-forwardable-flag-instead-of-denying-request.patch
 Patch117: Add-dns_canonicalize_hostname-fallback-support.patch
 
+Patch1000: krb5-kuserok-scripts.patch
+
 License: MIT
 URL: http://web.mit.edu/kerberos/www/
 BuildRequires: autoconf, bison, cmake, flex, gawk, gettext, pkgconfig, sed
@@ -168,6 +170,7 @@
 
 %package libs
 Summary: The non-admin shared libraries used by Kerberos 5
+Provides: scripts-krb5-libs, scripts-krb5-libs%{?_isa}
 Requires: coreutils, gawk, grep, sed
 Requires: keyutils-libs >= 1.5.8
 Requires: /etc/crypto-policies/back-ends/krb5.config
