--- krb5.spec.orig	2009-01-01 21:06:35.000000000 -0500
+++ krb5.spec	2009-01-01 22:53:25.000000000 -0500
@@ -16,7 +16,7 @@
 Summary: The Kerberos network authentication system.
 Name: krb5
 Version: 1.6.3
-Release: 10%{?dist}
+Release: 10%{?dist}.scripts.%{scriptsversion}
 # Maybe we should explode from the now-available-to-everybody tarball instead?
 # http://web.mit.edu/kerberos/dist/krb5/1.6/krb5-1.6.2-signed.tar
 Source0: krb5-%{version}.tar.gz
@@ -101,6 +101,8 @@
 Patch76: krb5-CVE-2007-5901.patch
 Patch77: krb5-CVE-2007-5971.patch
 
+Patch1000: krb5-kuserok-scripts.patch
+
 License: MIT, freely distributable.
 URL: http://web.mit.edu/kerberos/www/
 Group: System Environment/Libraries
@@ -142,6 +144,7 @@
 Group: System Environment/Libraries
 Prereq: grep, /sbin/ldconfig, sh-utils
 Obsoletes: krb5-configs
+Provides: scripts-krb5-libs
 
 %description libs
 Kerberos is a network authentication system. The krb5-libs package
@@ -1349,6 +1352,7 @@
 %patch75 -p0 -b .2008-0947
 %patch76 -p0 -b .2007-5901
 %patch77 -p0 -b .2007-5971
+%patch1000 -p1 -b .kuserok
 cp src/krb524/README README.krb524
 gzip doc/*.ps
 
