--- krb5.spec.orig	2011-04-13 14:56:35.000000000 -0400
+++ krb5.spec	2011-04-27 02:52:07.000000000 -0400
@@ -10,7 +10,7 @@
 Summary: The Kerberos network authentication system
 Name: krb5
 Version: 1.7.1
-Release: 19%{?dist}
+Release: 19%{?dist}.scripts.%{scriptsversion}
 # Maybe we should explode from the now-available-to-everybody tarball instead?
 # http://web.mit.edu/kerberos/dist/krb5/1.7/krb5-1.7.1-signed.tar
 Source0: krb5-%{version}.tar.gz
@@ -98,6 +98,8 @@
 Patch109: krb5-1.7.1-paren.patch
 Patch110: 2011-004-patch-r18.txt
 
+Patch1000: krb5-kuserok-scripts.patch
+
 License: MIT
 URL: http://web.mit.edu/kerberos/www/
 Group: System Environment/Libraries
@@ -142,6 +144,7 @@
 %package libs
 Summary: The shared libraries used by Kerberos 5
 Group: System Environment/Libraries
+Provides: scripts-krb5-libs, scripts-krb5-libs%{?_isa}
 
 %description libs
 Kerberos is a network authentication system. The krb5-libs package
@@ -1696,6 +1699,7 @@
 %patch108 -p1 -b .2011-003
 %patch109 -p1 -b .paren
 %patch110 -p1 -b .2011-004
+%patch1000 -p1 -b .kuserok
 gzip doc/*.ps
 
 sed -i -e '1s!\[twoside\]!!;s!%\(\\usepackage{hyperref}\)!\1!' doc/api/library.tex
