--- libpng.spec	2010-03-15 13:26:15.000000000 -0400
+++ libpng.spec	2010-07-14 00:07:10.000000000 -0400
@@ -1,8 +1,8 @@
 Summary: A library of functions for manipulating PNG image format files
 Name: libpng
 Epoch: 2
-Version: 1.2.43
-Release: 1%{?dist}
+Version: 1.2.44
+Release: 1.scripts.%{scriptsversion}%{?dist}
 License: zlib
 Group: System Environment/Libraries
 URL: http://www.libpng.org/pub/png/
@@ -94,6 +94,10 @@
 rm -rf $RPM_BUILD_ROOT
 
 %changelog
+* Tue Jul 13 2010 Geoffrey Thomas <geofft@mit.edu> 2:1.2.44-1
+- Update to libpng 1.2.44, includes fixes for CVE-2010-1205 and CVE-2010-2249
+  (This is equivalent to Tom Lane's 2:1.2.44-1 change from non-EOL'd distros)
+
 * Mon Mar 15 2010 Tom Lane <tgl@redhat.com> 2:1.2.43-1
 - Update to libpng 1.2.43, includes fix for CVE-2010-0205
 Related: #566234
