--- mit-zephyr.spec	2006-08-10 10:32:01.000000000 -0400
+++ mit-zephyr.spec.new	2007-01-14 15:43:04.000000000 -0500
@@ -67,7 +67,6 @@
 Source1: zhm.init
 URL: http://web.mit.edu/afs/dev.mit.edu/source/src-current/athena/lib/zephyr/
 Buildroot: %{_tmppath}/%{name}-%{version}-%{release}-buildroot
-Requires: mit-krb-config
 BuildPreReq: readline-devel, ncurses-devel
 %if %{redhat}=="yes"
 Requires: hesiod, krb5-libs
