--- rpmbuild/SPECS/mod_fcgid.spec.orig	2010-10-27 16:46:40.000000000 -0400
+++ rpmbuild/SPECS/mod_fcgid.spec	2010-10-27 17:47:10.000000000 -0400
@@ -15,13 +15,13 @@
 %endif
 
 Name:		mod_fcgid
-Version:	2.3.5
-Release:	1%{?dist}
+Version:	2.3.6.dev.1028094
+Release:	1%{?dist}.%{scriptsversion}
 Summary:	FastCGI interface module for Apache 2
 Group:		System Environment/Daemons
 License:	ASL 2.0
 URL:		http://httpd.apache.org/mod_fcgid/
-Source0:	http://www.apache.org/dist/httpd/mod_fcgid/mod_fcgid-%{version}.tar.bz2
+Source0:	http://web.mit.edu/scripts/deploy/mod_fcgid-%{version}.tar.bz
 Source1:	fcgid.conf
 Source2:	mod_fcgid-2.1-README.RPM
 Source3:	mod_fcgid-2.1-README.SELinux
