--- nss-ldapd.spec.orig	2009-03-30 04:33:50.000000000 -0400
+++ nss-ldapd.spec	2009-05-24 18:39:15.000000000 -0400
@@ -19,18 +19,19 @@
 
 
 Name:           nss-ldapd
-BuildRequires:  db-devel krb5-devel openldap2-devel
+BuildRequires:  db4-devel krb5-devel openldap-devel autoconf automake libtool
+Conflicts:      nss_ldap
 License:        LGPL v2.1 or later
 Group:          Productivity/Networking/LDAP/Clients
 AutoReqProv:    on
 PreReq:         /bin/chmod
 Conflicts:      nss_ldap
 Version:        0.6.8
-Release:        2.1
+Release:        2.1.scripts.%{scriptsversion}
 Summary:        NSS module and daemon for using LDAP as a naming service
 Url:            http://ch.tudelft.nl/~arthur/nss-ldapd/
 Source:         nss-ldapd-%{version}.tar.bz2
-Source1:        rc.nslcd
+Patch1000:      rc.nslcd.patch
 Source100:      nss-ldapd-0.6.8-rpmlintrc
 BuildRoot:      %{_tmppath}/%{name}-%{version}-build
 
@@ -52,7 +53,7 @@
 
 %prep
 %setup -q
-cp -v %{S:1} .
+%patch1000 -p0 -b .rc
 
 %build
 %{?suse_update_config:%{suse_update_config -f}}
@@ -72,7 +73,6 @@
 mkdir -p $RPM_BUILD_ROOT/etc/init.d/
 mkdir -p $RPM_BUILD_ROOT/usr/sbin/
 install -m 755 rc.nslcd $RPM_BUILD_ROOT/etc/init.d/nslcd
-ln -sf ../../etc/init.d/nslcd $RPM_BUILD_ROOT/usr/sbin/rcnslcd
 make DESTDIR=$RPM_BUILD_ROOT install
 mkdir -p $RPM_BUILD_ROOT/var/run/nslcd
 install -m 644 man/nss-ldapd.conf.5 $RPM_BUILD_ROOT/usr/share/man/man5
@@ -88,13 +88,8 @@
     /bin/chmod 640 /etc/nss-ldapd.conf
 fi
 
-%preun
-%stop_on_removal nslcd
-
 %postun 
 /sbin/ldconfig
-%restart_on_update nslcd
-%insserv_cleanup
 
 %files
 %defattr(-,root,root)
@@ -104,11 +99,12 @@
 %doc %{_mandir}/man8/*
 %config(noreplace) %attr(640,root,root) /etc/nss-ldapd.conf
 %config /etc/init.d/nslcd
-/usr/sbin/rcnslcd
 %dir /var/run/nslcd
 /usr/sbin/nslcd
 
 %changelog
+* Sun May 24 2009 Anders Kaseorg <andersk@mit.edu> - 0.6.8-2.1
+- Update for Fedora again.
 * Wed Mar 25 2009 rhafer@suse.de
 - Updated to 0.6.8:
   * the nss-ldapd.conf was created world-readable which could cause
@@ -129,6 +125,10 @@
 - Adjust config file permissions upon update, to fix world-readable
   /etc/nss-ldapd.conf as created by older versions
   (bnc#487737, CVE-2009-1073)
+* Wed Dec 31 2008  <quentin@mit.edu> - 0.6.4-3.0
+- add init script for Fedora
+* Wed Dec 31 2008  <quentin@mit.edu> - 0.6.4-2.4
+- port from openSUSE to Fedora
 * Fri Aug 15 2008 rhafer@suse.de
 - Fixed "Required-Stop" Tag to include the same services as
   "Required-Start"
