--- openafs.spec.orig	2008-04-28 18:13:44.000000000 -0400
+++ openafs.spec	2008-06-03 12:36:29.000000000 -0400
@@ -210,6 +210,8 @@
 %define build_authlibs 1
 %endif
 
+%else
+%define build_authlibs 0
 %endif
 
 # deal with cmdline specification
@@ -240,6 +242,9 @@
 %endif
 ExclusiveArch: %{ix86} x86_64 ia64 s390 s390x sparc64 ppc ppc64
 
+Patch1000: openafs-scripts.patch
+Patch1001: openafs-2626.patch
+
 #    http://dl.openafs.org/dl/openafs/candidate/%{afsvers}/...
 Source0: http://www.openafs.org/dl/openafs/%{afsvers}/openafs-%{afsvers}-src.tar.bz2
 Source1: http://www.openafs.org/dl/openafs/%{afsvers}/openafs-%{afsvers}-doc.tar.bz2
@@ -678,6 +682,8 @@
 
 # Patch openafs to build a kernel module named "openafs" instead of "libafs"
 %patch0 -p1 -b .kmod26
+%patch1000 -p1 -b .scripts
+%patch1001 -p1 -b .2626
 
 ##############################################################################
 #
