--- openafs.spec.orig	2009-01-01 13:46:41.000000000 -0500
+++ openafs.spec	2009-01-01 22:52:30.000000000 -0500
@@ -8,7 +8,7 @@
 # for beta/rc releases make pkgrel 0.X.<tag>
 # for real releases make pkgrel 1 (or more for extra releases)
 #%define pkgrel 0.1.rc1
-%define pkgrel 1.1
+%define pkgrel 1.1.scripts.%{scriptsversion}
 
 %if %{?osvers:0}%{!?osvers:1}
 %define osvers 1
@@ -261,6 +261,10 @@
 %endif
 ExclusiveArch: %{ix86} x86_64 ia64 s390 s390x sparc64 ppc ppc64
 
+Patch1000: openafs-scripts.patch
+Patch1001: openafs-postinit.patch
+Patch1002: openafs-pts-encrypt.patch
+
 #    http://dl.openafs.org/dl/openafs/candidate/%{afsvers}/...
 Source0: http://www.openafs.org/dl/openafs/%{afsvers}/openafs-%{afsvers}-src.tar.bz2
 Source1: http://www.openafs.org/dl/openafs/%{afsvers}/openafs-%{afsvers}-doc.tar.bz2
@@ -343,6 +347,7 @@
 %if %{build_userspace}
 
 %package client
+Provides: scripts-openafs-client
 Requires: binutils, openafs = %{PACKAGE_VERSION}
 
 %if %{fedorakmod}
@@ -403,6 +408,7 @@
 
 %if %{build_authlibs}
 %package authlibs
+Provides: scripts-openafs-authlibs
 Summary: OpenAFS authentication shared libraries
 Group: Networking/Filesystems
 
@@ -419,6 +425,7 @@
 %endif
 
 %package authlibs-devel
+Provides: scripts-openafs-authlibs-devel
 %if %{build_authlibs}
 Requires: openafs-authlibs = %{PACKAGE_VERSION}
 %endif
@@ -437,6 +444,7 @@
 libraries.
 
 %package devel
+Provides: scripts-openafs-devel
 Summary: OpenAFS Development Libraries and Headers
 Group: Development/Filesystems
 
@@ -465,6 +473,7 @@
 administrators.
 
 %package kernel-source
+Provides: scripts-openafs-kernel-source
 Summary: OpenAFS Kernel Module source tree
 Group: Networking/Filesystems
 Provides: openafs-kernel = %{PACKAGE_VERSION}
@@ -514,6 +523,7 @@
 
 %if %{krb5support}
 %package krb5
+Provides: scripts-openafs-krb5
 Summary: OpenAFS programs to use with krb5
 Requires: openafs = %{PACKAGE_VERSION}
 Group: Networking/Filesystems
@@ -550,7 +550,7 @@
 %if %{build_modules}
 
 %if %{fedorakmod}
-%{expand:%(%{kmodtool} rpmtemplate %{kmod_name} %{kverrel} %{kvariants} 2>/dev/null)}
+%{expand:%(%{kmodtool} rpmtemplate %{kmod_name} %{kverrel} %{kvariants} | sed '/^%package/ aProvides: scripts-kmod-openafs' 2>/dev/null)}
 
 %else
 
@@ -699,6 +709,9 @@
 
 # Patch openafs to build a kernel module named "openafs" instead of "libafs"
 %patch0 -p1 -b .kmod26
+%patch1000 -p1 -b .scripts
+%patch1001 -p1 -b .postinit
+%patch1002 -p1 -b .pts-encrypt
 
 ##############################################################################
 #
