--- openssh.spec.orig	2013-07-15 21:14:52.452894092 -0400
+++ openssh.spec	2013-07-15 21:18:59.494168115 -0400
@@ -74,7 +74,7 @@
 Summary: An open source implementation of SSH protocol versions 1 and 2
 Name: openssh
 Version: %{openssh_ver}
-Release: %{openssh_rel}%{?dist}%{?rescue_rel}
+Release: %{openssh_rel}%{?dist}%{?rescue_rel}.scripts.%{scriptsversion}
 URL: http://www.openssh.com/portable.html
 #URL1: http://pamsshagentauth.sourceforge.net
 #Source0: ftp://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-%{version}.tar.gz
@@ -220,7 +220,7 @@
 Patch904: openssh-5.9p1-change-max-startups.patch
 # make sftp's libedit interface marginally multibyte aware (#841771)
 Patch908: openssh-5.9p1-sftp-multibyte.patch
-
+Patch1001: openssh-4.7p1-gssapi-name-in-env.patch
 
 License: BSD
 Group: Applications/Internet
@@ -271,6 +273,7 @@
 Group: Applications/Internet
 Requires: openssh = %{version}-%{release}
 Requires: fipscheck-lib%{_isa} >= 1.3.0
+Provides: scripts-openssh-server
 
 %package server
 Summary: An open source SSH server daemon
@@ -458,10 +461,12 @@
 %patch714 -p0 -b .null-xcrypt
 
 %patch800 -p1 -b .gsskex
-%patch801 -p1 -b .force_krb
+# Remove force_krb as it conflicts with admof
+# %patch801 -p1 -b .force_krb
 
 %patch900 -p1 -b .canohost
-%patch901 -p1 -b .kuserok
+# Remove the kuserok patch as it won't apply without patch800
+# %patch901 -p1 -b .kuserok
 %patch902 -p1 -b .man-moduli
 %patch903 -p1 -b .ipqos
 %patch904 -p1 -b .max-startups
@@ -471,6 +476,8 @@
 # Nothing here yet
 %endif
 
+%patch1001 -p1 -b .gssapi-env
+
 autoreconf
 pushd pam_ssh_agent_auth-%{pam_ssh_agent_ver}
 autoreconf
