--- openssh.spec.orig	2009-01-01 21:06:01.000000000 -0500
+++ openssh.spec	2009-01-01 22:52:02.000000000 -0500
@@ -63,7 +63,7 @@
 Summary: The OpenSSH implementation of SSH protocol versions 1 and 2
 Name: openssh
 Version: 5.1p1
-Release: 3%{?dist}%{?rescue_rel}
+Release: 3%{?dist}%{?rescue_rel}.scripts.%{scriptsversion}
 URL: http://www.openssh.com/portable.html
 #Source0: ftp://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-%{version}.tar.gz
 #Source1: ftp://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-%{version}.tar.gz.asc
@@ -74,6 +74,7 @@
 Source1: openssh-nukeacss.sh
 Source2: sshd.pam
 Source3: sshd.init
+Patch1000: openssh-5.0p1-multihomed.patch
 Patch0: openssh-5.1p1-redhat.patch
 Patch2: openssh-5.1p1-skip-initial.patch
 Patch3: openssh-3.8.1p1-krb5-config.patch
@@ -159,6 +160,7 @@
 Requires(post): chkconfig >= 0.9, /sbin/service
 Requires(pre): /usr/sbin/useradd
 Requires: pam >= 1.0.1-3
+Provides: scripts-openssh-server
 
 %package askpass
 Summary: A passphrase dialog for OpenSSH and X
@@ -232,6 +234,8 @@
 %patch63 -p1 -b .bannerlen
 %patch64 -p1 -b .compat-sessions
 
+%patch1000 -p1 -b .multihomed
+
 autoreconf
 
 %build
