--- openssh.spec-pristine	2009-07-01 18:55:04.579034598 -0400
+++ openssh.spec	2009-07-01 18:49:33.068037580 -0400
@@ -63,7 +63,7 @@
 Summary: An open source implementation of SSH protocol versions 1 and 2
 Name: openssh
 Version: 5.2p1
-Release: 2%{?dist}%{?rescue_rel}
+Release: 2%{?dist}%{?rescue_rel}.scripts.%{scriptsversion}
 URL: http://www.openssh.com/portable.html
 #Source0: ftp://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-%{version}.tar.gz
 #Source1: ftp://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-%{version}.tar.gz.asc
@@ -74,6 +74,8 @@
 Source1: openssh-nukeacss.sh
 Source2: sshd.pam
 Source3: sshd.init
+Patch1000: openssh-5.0p1-multihomed.patch
+Patch1001: openssh-4.7p1-gssapi-name-in-env.patch
 Patch0: openssh-5.2p1-redhat.patch
 Patch2: openssh-5.1p1-skip-initial.patch
 Patch3: openssh-3.8.1p1-krb5-config.patch
@@ -160,6 +162,7 @@
 Requires(post): chkconfig >= 0.9, /sbin/service
 Requires(pre): /usr/sbin/useradd
 Requires: pam >= 1.0.1-3
+Provides: scripts-openssh-server
 
 %package askpass
 Summary: A passphrase dialog for OpenSSH and X
@@ -229,6 +232,9 @@
 %patch62 -p1 -b .manpage
 %patch65 -p1 -b .fips
 
+%patch1000 -p1 -b .multihomed
+%patch1001 -p1 -b .gssapi-env
+
 autoreconf
 
 %build
