--- openafs.spec.orig	2015-06-25 05:18:38.000000000 -0400
+++ openafs.spec	2015-07-13 21:41:31.196173273 -0400
@@ -4,7 +4,7 @@
 %define pkgvers 1.8.3
 # for beta/rc releases make pkgrel 0.<tag>
 # for real releases make pkgrel 1 (or more for extra releases)
-%define pkgrel 1
+%define pkgrel 1.99.scripts.%{scriptsversion}
 
 %{!?fedorakmod: %define fedorakmod 1}
 %{!?build_dkmspkg: %define build_dkmspkg 1}
@@ -255,9 +255,14 @@
 %if %{build_modules}
 BuildRequires: kernel-devel
 %endif
+BuildRequires: libtool
 
 ExclusiveArch: %{ix86} x86_64 ia64 s390 s390x sparc64 ppc ppc64 aarch64
 
+Patch1000: openafs-scripts.patch
+Patch1003: openafs-systemd-csdb.patch
+%define _default_patch_fuzz 2
+
 #    http://dl.openafs.org/dl/openafs/candidate/%{afsvers}/...
 Source0: http://www.openafs.org/dl/openafs/%{afsvers}/openafs-%{afsvers}-src.tar.bz2
 Source1: http://www.openafs.org/dl/openafs/%{afsvers}/openafs-%{afsvers}-doc.tar.bz2
@@ -337,6 +342,7 @@
 %if %{build_userspace}
 
 %package client
+Provides: scripts-openafs-client
 Requires: binutils, openafs = %{version}
 %if 0%{?fedora} >= 15 || 0%{?rhel} >= 7
 Requires: systemd-units
@@ -388,6 +394,7 @@
 %package -n dkms-%{name}
 Summary:        DKMS-ready kernel source for AFS distributed filesystem
 Group:          Development/Kernel
+Provides:       scripts-dkms-%{name}
 Provides:       openafs-kernel = %{version}
 %if %{fedorakmod}
 Provides: %{name}-kmod = %{version}
@@ -409,6 +416,7 @@
 
 %if %{build_authlibs}
 %package authlibs
+Provides: scripts-openafs-authlibs
 Summary: OpenAFS authentication shared libraries
 Group: Networking/Filesystems
 
@@ -425,6 +433,7 @@
 %endif
 
 %package authlibs-devel
+Provides: scripts-openafs-authlibs-devel
 %if %{build_authlibs}
 Requires: openafs-authlibs = %{version}-%{release}
 %endif
@@ -443,6 +452,7 @@
 libraries.
 
 %package devel
+Provides: scripts-openafs-devel
 Summary: OpenAFS Development Libraries and Headers
 Group: Development/Filesystems
 Requires: openafs = %{version}-%{release}
@@ -472,6 +482,7 @@
 administrators.
 
 %package kernel-source
+Provides: scripts-openafs-kernel-source
 Summary: OpenAFS Kernel Module source tree
 Group: Networking/Filesystems
 Provides: openafs-kernel = %{version}
@@ -521,6 +532,7 @@
 
 %if %{krb5support}
 %package krb5
+Provides: scripts-openafs-krb5
 Summary: OpenAFS programs to use with krb5
 Requires: openafs = %{version}
 Group: Networking/Filesystems
@@ -547,6 +559,6 @@
 %if %{build_modules}
 
-%{expand:%(%{kmodtool} rpmtemplate %{kmod_name} %{kverrel} %{depmod} %{kvariants} 2>/dev/null)}
+%{expand:%(%{kmodtool} rpmtemplate %{kmod_name} %{kverrel} %{depmod} %{kvariants} | sed '/^%package/ aProvides: scripts-kmod-openafs' 2>/dev/null)}
 
 %endif
 
@@ -704,6 +716,10 @@
 #%setup -q -n %{srcdir}
 %setup -q -b 1 -n %{srcdir}
 
+# Apply the Scripts patch
+%patch1000 -p1 -b .scripts
+%patch1003 -p1 -b .systemd-csdb
+
 ##############################################################################
 #
 # building
@@ -877,6 +893,8 @@
 %endif
 %endif
 
+./regen.sh
+
 ./configure --with-afs-sysname=${sysname} \
        --prefix=%{_prefix} \
        --libdir=%{_libdir} \
@@ -746,6 +764,13 @@
 rm -f $RPM_BUILD_ROOT%{_libdir}/libafsauthent.so*
 rm -f $RPM_BUILD_ROOT%{_libdir}/libafsrpc.so*
 rm -f $RPM_BUILD_ROOT%{_libdir}/libkopenafs.so*
+%else
+chmod +x $RPM_BUILD_ROOT%{_libdir}/libafsauthent.so
+chmod +x $RPM_BUILD_ROOT%{_libdir}/libafsrpc.so
+chmod +x $RPM_BUILD_ROOT%{_libdir}/libafsauthent.so.*
+chmod +x $RPM_BUILD_ROOT%{_libdir}/libafsrpc.so.*
+chmod +x $RPM_BUILD_ROOT%{_libdir}/libkopenafs.so
+chmod +x $RPM_BUILD_ROOT%{_libdir}/libkopenafs.so.*
 %endif
 rm -f $RPM_BUILD_ROOT%{_sbindir}/afsd.fuse
 %if !%{kauth_support}
