--- php.spec.~1~	2012-04-27 11:35:35.000000000 -0400
+++ php.spec	2012-05-23 00:39:40.233431646 -0400
@@ -38,7 +38,7 @@
 Summary: PHP scripting language for creating dynamic web sites
 Name: php
 Version: 5.3.11
-Release: 1%{?dist}
+Release: 1%{?dist}.scripts.%{scriptsversion}
 License: PHP
 Group: Development/Languages
 URL: http://www.php.net/
@@ -78,6 +78,9 @@
 # Fixes for tests
 Patch61: php-5.0.4-tests-wddx.patch
 
+Provides: scripts-php
+Patch100: php-cve-2012-1823.patch
+
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
 BuildRequires: bzip2-devel, curl-devel >= 7.9, db4-devel, gmp-devel
@@ -546,6 +549,8 @@
 
 %patch61 -p1 -b .tests-wddx
 
+%patch100 -p0 -b .cve-2012-1823
+
 # Prevent %%doc confusion over LICENSE files
 cp Zend/LICENSE Zend/ZEND_LICENSE
 cp TSRM/LICENSE TSRM_LICENSE
