--- php.spec.~1~	2012-04-27 13:42:25.000000000 -0400
+++ php.spec	2012-05-22 23:38:56.360148746 -0400
@@ -56,7 +56,7 @@
 Summary: PHP scripting language for creating dynamic web sites
 Name: php
 Version: 5.4.1
-Release: 1%{?dist}
+Release: 1%{?dist}.scripts.%{scriptsversion}
 License: PHP
 Group: Development/Languages
 URL: http://www.php.net/
@@ -91,6 +91,9 @@
 # Fixes for tests
 
 
+Provides: scripts-php
+Patch100: php-cve-2012-1823.patch
+
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
 BuildRequires: bzip2-devel, curl-devel >= 7.9, %{db_devel}, gmp-devel
@@ -584,6 +587,8 @@
 %patch44 -p1 -b .systzip
 %endif
 
+%patch100 -p0 -b .cve-2012-1823
+
 # Prevent %%doc confusion over LICENSE files
 cp Zend/LICENSE Zend/ZEND_LICENSE
 cp TSRM/LICENSE TSRM_LICENSE
