--- python-flask.spec.orig	2011-02-26 15:09:18.000000000 -0500
+++ python-flask.spec	2011-11-10 22:14:25.926527870 -0500
@@ -4,7 +4,7 @@
 
 Name:           python-flask
 Version:        0.6.1
-Release:        2%{?dist}
+Release:        2%{?dist}.scripts.%{scriptsversion}
 Summary:        A micro-framework for Python based on Werkzeug, Jinja 2 and good intentions
 
 Group:          Development/Libraries
@@ -59,6 +59,7 @@
 
 rm -rf $RPM_BUILD_ROOT%{python_sitelib}/site.py
 rm -rf $RPM_BUILD_ROOT%{python_sitelib}/site.py[co]
+rm -rf $RPM_BUILD_ROOT%{python_sitelib}/easy-install.pth
 rm -rf docs/_build/html/.buildinfo
 rm -rf examples/minitwit/*.pyc
 rm -rf examples/flaskr/*.pyc
@@ -73,7 +74,9 @@
 %files
 %defattr(-,root,root,-)
 %doc AUTHORS LICENSE PKG-INFO CHANGES README
-%{python_sitelib}/*
+%{python_sitelib}/*.egg-info
+%{python_sitelib}/*.egg-link
+%{python_sitelib}/flask
 
 %files doc
 %defattr(-,root,root,-)
