--- redland-bindings.spec.~1~	2012-03-06 07:34:14.000000000 -0500
+++ redland-bindings.spec	2012-05-22 23:43:49.839142648 -0400
@@ -1,6 +1,6 @@
 Name:           redland-bindings
 Version:        1.0.14.1
-Release:        1%{?dist}
+Release:        1%{?dist}.scripts.%{scriptsversion}
 Summary:        Redland RDF Application Framework API Bindings
 
 Group:          System Environment/Libraries
@@ -130,6 +130,8 @@
 #autoconf
 #automake
 
+# Horrible hack to make PHP work
+sed -i -e 's/zend_error_noreturn/zend_error/' php/*
 
 %build
 %configure \
