--- redland-bindings.spec.old	2011-10-12 14:51:58.000000000 -0400
+++ redland-bindings.spec	2011-10-13 10:15:24.000000000 -0400
@@ -1,6 +1,6 @@
 Name:           redland-bindings
 Version:        1.0.11.1
-Release:        6%{?dist}
+Release:        6%{?dist}.scripts.%{scriptsversion}
 Summary:        Redland RDF Application Framework API Bindings
 
 Group:          System Environment/Libraries
@@ -137,6 +137,9 @@
 autoconf
 automake
 
+# Horrible hack to make PHP work
+sed -i -e 's/zend_error_noreturn/zend_error/' php/*
+
 # when specifying system redland, the configure check for the version
 # allows newer redland libraries; without it it doesn't.
 # authors says that older bindings work with newer redlands.
