--- rubygems.spec.orig	2013-04-05 15:02:08.147935906 -0400
+++ rubygems.spec	2013-04-05 15:04:14.512566960 -0400
@@ -26,7 +26,7 @@
 Summary:	The Ruby standard for packaging ruby libraries
 Name:		rubygems
 Version:	1.8.25
-Release:	6%{?dist}
+Release:	6%{?dist}.scripts.%{scriptsversion}
 Group:		Development/Libraries
 License:	Ruby or MIT
 
@@ -60,6 +60,8 @@
 # https://github.com/rubygems/rubygems/issues/210
 Patch109:	rubygems-1.8.11-binary-extensions.patch
 
+Provides: scripts-rubygems = %{version}-%{release}
+Patch1000: rubygems-rails-require-thread.patch
 
 Requires:	ruby(abi) = 1.9.1
 Requires:	rubygem(rdoc) >= 3.9.4
@@ -105,6 +107,8 @@
 %patch105 -p1 -b .uninst
 %patch109 -p1 -b .bindir
 
+%patch1000 -p1 -b .thread
+
 # Some of the library files start with #! which rpmlint doesn't like
 # and doesn't make much sense
 for f in `find lib -name \*.rb` ; do
