--- rubygem-rails.spec.orig	2009-07-18 02:15:11.040937830 -0400
+++ rubygem-rails.spec	2009-07-18 02:21:24.665129305 -0400
@@ -7,7 +7,7 @@
 Summary: Web-application framework
 Name: rubygem-%{gemname}
 Version: 2.3.2
-Release: 2%{?dist}
+Release: 2%{?dist}.scripts.%{scriptsversion}
 Group: Development/Languages
 License: MIT
 URL: http://www.rubyonrails.org
@@ -23,6 +23,7 @@
 BuildRequires: rubygems
 BuildArch: noarch
 Provides: rubygem(%{gemname}) = %{version}
+Provides: scripts-rubygem-rails
 
 %description
 Rails is a framework for building web-application using CGI, FCGI, mod_ruby,
@@ -55,9 +56,6 @@
 # Remove backup files
 find %{buildroot}/%{geminstdir} -type f -name "*~" -delete
 
-# Delete zero-length files
-find %{buildroot}/%{geminstdir} -type f -size 0c -exec rm -rvf {} \;
-
 # Fix anything executable that does not have a shebang
 for file in `find %{buildroot}/%{geminstdir} -type f -perm /a+x`; do
     [ -z "`head -n 1 $file | grep \"^#!/\"`" ] && chmod -v 644 $file
@@ -100,6 +98,9 @@
 
 
 %changelog
+* Sat Jul 18 2009 Mitchell Berger <mitchb@mit.edu> - 2.3.2-2.scripts
+- Do not remove required zero-length files
+
 * Wed May  6 2009 David Lutterkort <lutter@redhat.com> - 2.3.2-2
 - Fix replacement of shebang lines; broke scripts/generate (bz 496480)
 
