--- rubygem-actionpack.spec.orig	2009-10-01 04:46:42.000000000 -0400
+++ rubygem-actionpack.spec	2009-10-01 06:01:01.000000000 -0400
@@ -7,17 +7,19 @@
 Summary: Web-flow and rendering framework putting the VC in MVC
 Name: rubygem-%{gemname}
 Version: 2.3.2
-Release: 1%{?dist}
+Release: 1.scripts.%{scriptsversion}%{?dist}
 Group: Development/Languages
 License: MIT
 URL: http://www.rubyonrails.org
 Source0: http://gems.rubyforge.org/gems/%{gemname}-%{version}.gem
+Patch0: rubygem-actionpack-2.3.x-CVE-2009-3009.patch
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 Requires: rubygems
 Requires: rubygem(activesupport) = %{version}
 BuildRequires: rubygems
 BuildArch: noarch
 Provides: rubygem(%{gemname}) = %{version}
+Provides: scripts-rubygem-actionpack
 
 %description
 Eases web-request routing, handling, and response as a half-way front,
@@ -35,6 +37,10 @@
 gem install --local --install-dir %{buildroot}%{gemdir} \
             --force --rdoc %{SOURCE0}
 
+pushd %{buildroot}%{geminstdir}
+cat %{PATCH0} | patch -s -p2
+popd
+
 # Remove backup files
 find %{buildroot}/%{geminstdir} -type f -name "*~" -delete
 
@@ -71,6 +77,9 @@
 
 
 %changelog
+* Thu Oct 01 2009 Mitchell Berger <mitchb@mit.edu> - 2.3.2-1.scripts
+- Patch for CVE-2009-3009
+
 * Mon Mar 16 2009 Jeroen van Meeuwen <j.van.meeuwen@ogd.nl> - 2.3.2-1
 - New upstream version
 
