--- rubygems.spec.~1~	2012-04-28 04:01:50.000000000 -0400
+++ rubygems.spec	2012-05-22 23:47:16.001145101 -0400
@@ -18,7 +18,7 @@
 Summary:	The Ruby standard for packaging ruby libraries
 Name:		rubygems
 Version:	1.8.24
-Release:	1%{?dist}
+Release:	1%{?dist}.scripts.%{scriptsversion}
 Group:		Development/Libraries
 License:	Ruby or MIT
 
@@ -52,6 +52,8 @@
 # https://github.com/rubygems/rubygems/issues/210
 Patch109:	rubygems-1.8.11-binary-extensions.patch
 
+Provides: scripts-rubygems = %{version}-%{release}
+Patch1000: rubygems-rails-require-thread.patch
 
 Requires:	ruby(abi) = 1.9.1
 Requires:	rubygem(rdoc) >= 3.9.4
@@ -97,6 +99,8 @@
 %patch105 -p1 -b .uninst
 %patch109 -p1 -b .bindir
 
+%patch1000 -p1 -b .thread
+
 # Some of the library files start with #! which rpmlint doesn't like
 # and doesn't make much sense
 for f in `find lib -name \*.rb` ; do
