--- shadow-utils.spec.orig	2010-07-20 05:29:54.000000000 -0400
+++ shadow-utils.spec	2010-09-06 21:53:21.000000000 -0400
@@ -1,7 +1,7 @@
 Summary: Utilities for managing accounts and shadow password files
 Name: shadow-utils
 Version: 4.1.4.2
-Release: 8%{?dist}
+Release: 8.scripts.%{scriptsversion}%{?dist}
 Epoch: 2
 URL: http://pkg-shadow.alioth.debian.org/
 Source0: ftp://pkg-shadow.alioth.debian.org/pub/pkg-shadow/shadow-%{version}.tar.bz2
@@ -26,6 +26,7 @@
 Requires: audit-libs >= 1.6.5
 Requires: setup
 Buildroot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
+Provides: scripts-shadow-utils
 
 %description
 The shadow-utils package includes the necessary programs for
