--- openafs-client.init	2007-01-14 01:49:13.288731294 -0500
+++ openafs-client.init	2007-01-14 01:49:48.044846780 -0500
@@ -39,6 +39,8 @@
 	fi
 	/usr/vice/etc/afsd $AFSD_ARGS
 	RETVAL=$?
+	/usr/bin/fs setcrypt on
+	/usr/bin/fs sysname 'i386_rhel3' 'i386_rh9' 'i386_linux24' 'i386_linux22' 'i386_linux3' 'i386_linux2' 'i386_linux1'
 	echo
 	[ $RETVAL -eq 0 ] && touch /var/lock/subsys/openafs-client
 	return $RETVAL
