diff --git a/ldap/servers/plugins/replication/cl5_api.c b/ldap/servers/plugins/replication/cl5_api.c
index b7c7229..59ef8be 100644
--- a/ldap/servers/plugins/replication/cl5_api.c
+++ b/ldap/servers/plugins/replication/cl5_api.c
@@ -657,7 +657,7 @@ int cl5DeleteDBSync (Object *replica)
         file = (CL5DBFile*)object_get_data (obj);
         PR_ASSERT (file);
 
-        PR_snprintf (fName, MAXPATHLEN, "%s/%s", s_cl5Desc.dbDir, file->name);
+        PR_snprintf (fName, MAXPATHLEN, "%s", file->name);
         
         _cl5DBDeleteFile (obj);
 
@@ -1894,6 +1894,7 @@ static int _cl5DBOpen ()
                 slapi_log_error(SLAPI_LOG_REPL, repl_plugin_name_cl, "_cl5DBOpen: "
                           "file %s has no matching replica; removing\n", entry->name);
 
+                PR_snprintf(fullpathname, MAXPATHLEN, "%s/%s", s_cl5Desc.dbDir, entry->name);
                 rc = s_cl5Desc.dbEnv->dbremove(s_cl5Desc.dbEnv,
                                                0, fullpathname, 0, 0);
                 if (rc != 0)
@@ -5528,7 +5529,6 @@ done:
 static void _cl5DBCloseFile (void **data)
 { 
 	CL5DBFile *file;
-    char fullpathname[MAXPATHLEN];
                 				
 	PR_ASSERT (data);
 
@@ -5537,7 +5537,7 @@ static void _cl5DBCloseFile (void **data)
 	PR_ASSERT (file);
 
 	slapi_log_error(SLAPI_LOG_REPL, repl_plugin_name_cl, "_cl5DBCloseFile: "
-					"Closing database %s/%s\n", s_cl5Desc.dbDir, file->name);
+					"Closing database %s\n", file->name);
 
 	/* close the file */
 	/* if this is normal close or close after import, update entry count */	
@@ -5553,7 +5553,7 @@ static void _cl5DBCloseFile (void **data)
 	if (file->db) {
 	    file->db->close(file->db, 0);
 	    slapi_log_error(SLAPI_LOG_REPL, repl_plugin_name_cl, "_cl5DBCloseFile: "
-						"Closed the changelog database handle for %s/%s\n", s_cl5Desc.dbDir, file->name);
+						"Closed the changelog database handle for %s\n", file->name);
 	    file->db = NULL;
 	}
 
@@ -5562,16 +5562,15 @@ static void _cl5DBCloseFile (void **data)
 		int rc = 0;
 		/* We need to use the libdb API to delete the files, otherwise we'll
 		 * run into problems when we try to checkpoint transactions later. */
-		PR_snprintf(fullpathname, MAXPATHLEN, "%s/%s", s_cl5Desc.dbDir, file->name);
-		rc = s_cl5Desc.dbEnv->dbremove(s_cl5Desc.dbEnv, 0, fullpathname, 0, 0);
+		rc = s_cl5Desc.dbEnv->dbremove(s_cl5Desc.dbEnv, 0, file->name, 0, 0);
 		if (rc != 0)
 		{
 			slapi_log_error(SLAPI_LOG_REPL, repl_plugin_name_cl, "_cl5DBCloseFile: "
 							"failed to remove (%s) file; libdb error - %d (%s)\n", 
-							fullpathname, rc, db_strerror(rc));
+							file->name, rc, db_strerror(rc));
 		} else {
 			slapi_log_error(SLAPI_LOG_REPL, repl_plugin_name_cl, "_cl5DBCloseFile: "
-							"Deleted the changelog database file %s/%s\n", s_cl5Desc.dbDir, file->name);
+							"Deleted the changelog database file %s\n", file->name);
 
         }
 	}
