diff --git a/variables.c b/variables.c
index cdc54bc..74f3a8c 100644
--- a/variables.c
+++ b/variables.c
@@ -349,7 +349,7 @@ initialize_shell_variables (env, privmode)
 
       /* If exported function, define it now.  Don't import functions from
 	 the environment in privileged mode. */
-      if (privmode == 0 && read_but_dont_execute == 0 && STREQN ("() {", string, 4))
+      if (0)
 	{
 	  string_length = strlen (string);
 	  temp_string = (char *)xmalloc (3 + string_length + char_index);
