--- a/httpd.init.orig	2011-11-05 19:10:18.897425517 -0400
+++ b/httpd.init	2011-11-05 19:11:22.523201035 -0400
@@ -12,7 +12,7 @@
 #
 ### BEGIN INIT INFO
 # Provides: httpd
-# Required-Start: $local_fs $remote_fs $network $named
+# Required-Start: $local_fs $remote_fs $network $named openafs-client crond
 # Required-Stop: $local_fs $remote_fs $network
 # Should-Start: distcache
 # Short-Description: start and stop Apache HTTP Server
