diff --git a/src/packaging/RedHat/openafs-client.modules b/src/packaging/RedHat/openafs-client.modules
new file mode 100644
index 0000000..055d117
--- /dev/null
+++ b/src/packaging/RedHat/openafs-client.modules
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+# Load the OpenAFS kernel module at boot
+exec /sbin/modprobe openafs
diff --git a/src/packaging/RedHat/openafs-client.service b/src/packaging/RedHat/openafs-client.service
new file mode 100644
index 0000000..bc95057
--- /dev/null
+++ b/src/packaging/RedHat/openafs-client.service
@@ -0,0 +1,18 @@
+[Unit]
+Description=OpenAFS Client Service
+After=syslog.target network.target
+
+[Service]
+Type=forking
+EnvironmentFile=/etc/sysconfig/openafs
+ExecStartPre=/bin/sed -n 'w/usr/vice/etc/CellServDB' /usr/vice/etc/CellServDB.local /usr/vice/etc/CellServDB.dist
+ExecStartPre=/bin/chmod 0644 /usr/vice/etc/CellServDB
+ExecStartPre=/sbin/modprobe openafs
+ExecStart=/usr/vice/etc/afsd $AFSD_ARGS
+ExecStop=/bin/umount /afs
+ExecStop=/usr/vice/etc/afsd -shutdown
+ExecStop=/sbin/rmmod openafs
+KillMode=none
+
+[Install]
+WantedBy=multi-user.target
diff --git a/src/packaging/RedHat/openafs-server.service b/src/packaging/RedHat/openafs-server.service
new file mode 100644
index 0000000..2d34bb0
--- /dev/null
+++ b/src/packaging/RedHat/openafs-server.service
@@ -0,0 +1,11 @@
+[Unit]
+Description=OpenAFS Server Service
+After=syslog.target network.target
+
+[Service]
+EnvironmentFile=-/etc/sysconfig/openafs
+ExecStart=/usr/afs/bin/bosserver $BOSSERVER_ARGS
+ExecStop=/usr/bin/bos shutdown localhost -wait -localauth
+
+[Install]
+WantedBy=multi-user.target
