diff --git a/gss-serv.c b/gss-serv.c
index 1c0ac53..69bf73d 100644
--- a/gss-serv.c
+++ b/gss-serv.c
@@ -441,6 +441,18 @@ ssh_gssapi_do_child(char ***envp, u_int *envsizep)
 	}
 }
 
+void
+ssh_gssapi_do_child_principalname(char ***envp, u_int *envsizep)
+{
+	if (gssapi_client.exportedname.length != 0 &&
+	    gssapi_client.exportedname.value != NULL) {
+	        debug("Setting %s to %s", "SSH_GSSAPI_NAME",
+		    gssapi_client.exportedname.value);
+		child_set_env(envp, envsizep, "SSH_GSSAPI_NAME",
+		    gssapi_client.exportedname.value);
+	}
+}
+
 /* Privileged */
 int
 ssh_gssapi_userok(char *user, struct passwd *pw, int kex)
diff --git a/session.c b/session.c
index b989afc..f908d0e 100644
--- a/session.c
+++ b/session.c
@@ -1085,6 +1085,7 @@ do_setup_env(struct ssh *ssh, Session *s, const char *shell)
 	 */
 	if (s->authctxt->krb5_set_env)
 		ssh_gssapi_do_child(&env, &envsize);
+	ssh_gssapi_do_child_principalname(&env, &envsize);
 #endif
 
 	/* Set basic environment. */
diff --git a/ssh-gss.h b/ssh-gss.h
index c7ec22d..e865d1e 100644
--- a/ssh-gss.h
+++ b/ssh-gss.h
@@ -170,6 +170,7 @@ OM_uint32 ssh_gssapi_server_ctx(Gssctxt **, gss_OID);
 int ssh_gssapi_userok(char *name, struct passwd *, int kex);
 OM_uint32 ssh_gssapi_checkmic(Gssctxt *, gss_buffer_t, gss_buffer_t);
 void ssh_gssapi_do_child(char ***, u_int *);
+void ssh_gssapi_do_child_principalname(char ***, u_int *);
 void ssh_gssapi_cleanup_creds(void);
 int ssh_gssapi_storecreds(void);
 const char *ssh_gssapi_displayname(void);
