diff --git a/src/packaging/RedHat/openafs-client.init b/src/packaging/RedHat/openafs-client.init
index e8128aa..10ec647 100644
--- a/src/packaging/RedHat/openafs-client.init
+++ b/src/packaging/RedHat/openafs-client.init
@@ -37,6 +37,7 @@ start() {
 	RETVAL=$?
 	echo
 	[ $RETVAL -eq 0 ] && touch /var/lock/subsys/openafs-client
+	[ $RETVAL -eq 0 ] && [ "$AFS_POST_INIT" ] && $AFS_POST_INIT
 	return $RETVAL
 }
 
