--- openafs-1.4.10/src/afs/afs.h.orig	2009-06-06 21:02:53.000000000 -0400
+++ openafs-1.4.10/src/afs/afs.h	2009-06-06 21:03:28.000000000 -0400
@@ -75,7 +75,7 @@
 #define	PIGGYSIZE	1350	/* max piggyback size */
 #define	MAXVOLS		128	/* max vols we can store */
 #define	MAXSYSNAME	128	/* max sysname (i.e. @sys) size */
-#define MAXNUMSYSNAMES	16	/* max that current constants allow */
+#define MAXNUMSYSNAMES	32	/* max that current constants allow */
 #define	NOTOKTIMEOUT	(2*3600)	/* time after which to timeout conns sans tokens */
 #define	NOPAG		0xffffffff
 #define AFS_NCBRS	300	/* max # of call back return entries */
