diff --git a/src/packaging/RedHat/openafs-client.service b/src/packaging/RedHat/openafs-client.service
index bc95057..9627280 100644
--- a/src/packaging/RedHat/openafs-client.service
+++ b/src/packaging/RedHat/openafs-client.service
@@ -1,5 +1,6 @@
 [Unit]
 Description=OpenAFS Client Service
+Before=crond.service
 After=syslog.target network.target
 
 [Service]
@@ -15,4 +16,4 @@ ExecStop=/sbin/rmmod openafs
 KillMode=none
 
 [Install]
-WantedBy=multi-user.target
+WantedBy=multi-user.target crond.service
